pkg/config/types.go

```diff
@@ -89,7 +100,20 @@ type Config struct {
 	// Kubernetes username + groups.
 	UserMappings []UserMapping
 
+	// NodeMappings is a list of mappings from AWS IAM Role to
+	// Kubernetes node name (i.e. system:node:NodeName).
+	NodeMappings []NodeMapping
+
 	// AutoMappedAWSAccounts is a list of AWS accounts that are allowed without an explicit user/role mapping.
 	// IAM ARN from these accounts automatically maps to the Kubernetes username.
 	AutoMappedAWSAccounts []string
+
+	// DefaultEC2DescribeInstancesRoleARN is an optional AWS Resource Name for an IAM Role to be assumed
+	// before calling ec2:DescribeInstances to determine the private DNS of the calling kubelet (EC2 Instance).
+	// If nil, defaults to using the IAM Role attached to the instance where heptio-authenticator-aws is
+	// running.
+	DefaultEC2DescribeInstancesRoleARN string
+
+	// Region is the AWS region, such as us-east-1.  If nil, discovered by calling ec2 metadata.
+	Region string
```

Comment:
This can be removed an inferred from EC2 instance metadata. If it's running on a non-EC2 instance like for development then it can be inferred from environment variables along with credentials.